beacon-chain/rpc/validator_server.go

```diff
@@ -30,36 +30,36 @@ type ValidatorServer struct {
 // beacon state, if not, then it creates a stream which listens for canonical states which contain
 // the validator with the public key as an active validator record.
 func (vs *ValidatorServer) WaitForActivation(req *pb.ValidatorActivationRequest, stream pb.ValidatorService_WaitForActivationServer) error {
-	if vs.beaconDB.HasValidator(req.Pubkey) {
-		beaconState, err := vs.beaconDB.HeadState(vs.ctx)
+	if vs.beaconDB.HasValidators(req.PublicKeys) {
+		beaconState, err := vs.beaconDB.HeadState(stream.Context())
 		if err != nil {
 			return fmt.Errorf("could not retrieve beacon state: %v", err)
 		}
-		activeVal, err := vs.retrieveActiveValidator(beaconState, req.Pubkey)
+		activeKeys, err := vs.filterActivePublicKeys(beaconState, req.PublicKeys)
 		if err != nil {
 			return fmt.Errorf("could not retrieve active validator from state: %v", err)
 		}
 		res := &pb.ValidatorActivationResponse{
-			Validator: activeVal,
+			ActivatedPublicKeys: activeKeys,
 		}
 		return stream.Send(res)
 	}
 	for {
 		select {
 		case <-time.After(3 * time.Second):
-			if !vs.beaconDB.HasValidator(req.Pubkey) {
+			if !vs.beaconDB.HasValidators(req.PublicKeys) {
```

Comment:
So this is not a good condition, right? If any of the validators doesn't exist, we don't perform anyone's role. Can you improve this logic?

Reply:
fixed by changing HasValidators logic